examples/server/server.cpp

```diff
@@ -2967,11 +2967,17 @@ int main(int argc, char ** argv) {
     };
 
     const auto handle_props = [&ctx_server](const httplib::Request & req, httplib::Response & res) {
+        std::vector<char> model_template(2048, 0); // longest known template is about 1200 bytes
+        std::string template_key = "tokenizer.chat_template", curr_tmpl;
+        if (llama_model_meta_val_str(ctx_server.model, template_key.c_str(), model_template.data(), model_template.size()) > 0) {
+            curr_tmpl = std::string(model_template.data(), model_template.size());
+        }
```

Comment:
This code is copied from `llama_chat_apply_template`, but there we use heuristic check so we don't need to retrieve full chat template (see `tmpl_contains` function). It still works even if the template is longer than 2048 bytes.

However, if you want to expose it to the frontend, it's better to pay attention to edge case (i.e. template is longer than 2048)